### PR TITLE
refactor: Change all screenshot implementations to use the low-level API call

### LINKED ISSUE #15199: [IOS] After test fail on real device , Appium is throwing error.  On simulator it works fine and continues  testing
## Environment

* Appium version (or git revision) that exhibits the issue:1.20.2
* Last Appium version that did not exhibit the issue (if applicable):
* Desktop OS/version used to run Appium: OS
* Node.js version (unless using Appium.app|exe):
* Npm or Yarn package manager:npm
* Mobile platform/version under test:14.4.1
* Real device or emulator/simulator:Real device
* Appium CLI or Appium.app|exe:

## Details

If necessary, describe the problem you have been experiencing in more detail.

## Link to Appium logs

https://gist.github.com/levank707/33ee1accbb66ba5f96796587c89ddb1c

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Fixes https://github.com/appium/appium/issues/15199

### CODE REVIEW

File: WebDriverAgentLib/Utilities/FBImageUtils.m [hunk 1]
```diff
@@ -45,61 +45,3 @@ BOOL FBIsPngImage(NSData *imageData)
...
-      imageOrientation = UIImageOrientationLeft;
-    } else if (orientation == UIInterfaceOrientationLandscapeLeft) {
-      imageOrientation = UIImageOrientationRight;
-    } else if (orientation == UIInterfaceOrientationPortraitUpsideDown) {
-      imageOrientation = UIImageOrientationDown;
-    } else {
-      return FBToPngData(screenshotData);
-    }
-  } else {
-    switch (FBConfiguration.screenshotOrientation) {
```
Comment (KazuCocoa) [inline WebDriverAgentLib/Utilities/FBImageUtils.m:76]: Can we keep this settings in https://github.com/appium/WebDriverAgent/pull/482/files#diff-50f20c000d3b1215568d23ae0d9e54468c4aa940353be843759688cd69571de2R129 to set the screenshot orientation forcefully?
This setting's necessity happens on real devices and iPad devices/simulators. (Especially landscape mode)

**Review by KazuCocoa — COMMENTED** [review]
👍 
i just tested this morning this. Streaming (9100 port) worked, btw.

File: WebDriverAgentLib/Utilities/FBImageUtils.m (hunk 1 above)
Comment (mykola-mokhnach) [reply WebDriverAgentLib/Utilities/FBImageUtils.m:76]: I tested the current implementation with different device orientations and it seems to be fine now for all of them. Could you please tell me which combination of device/orientation does not work there?

Comment (KazuCocoa) [reply WebDriverAgentLib/Utilities/FBImageUtils.m:76]: I don't have the device, but the original report I introduced this change was iOS 13.0 or around and iPhone 8(?) with a gaming app. A customer told me some iPad on their env as well. (But I don't remember these iPads' concrete versions)

btw, this case also I do not have a device, but previously I observed Xcode 10.3 had this wrong coordinate issue but Xcode 11.4 or around did not have it. (I forgot this exact environment...)
So, potentially newer Xcode versions fixed these issues.

Comment (mykola-mokhnach) [reply WebDriverAgentLib/Utilities/FBImageUtils.m:76]: added some code to enforce image orientation if requested

**Review by KazuCocoa — APPROVED** [review]
Seems good.
Thanks :)

File: WebDriverAgentLib/Utilities/FBImageIOScaler.m [hunk 2]
```diff
@@ -55,51 +65,105 @@ - (void)submitImage:(NSData *)image scalingFactor:(CGFloat)scalingFactor compres
...
+  } else if (FBConfiguration.screenshotOrientation == UIInterfaceOrientationPortraitUpsideDown) {
+    orientation = UIImageOrientationDown;
+  } else if (FBConfiguration.screenshotOrientation == UIInterfaceOrientationLandscapeLeft) {
+    orientation = UIImageOrientationRight;
+  } else if (FBConfiguration.screenshotOrientation == UIInterfaceOrientationLandscapeRight) {
+    orientation = UIImageOrientationLeft;
+  }
+#endif
+  uiImage = [UIImage imageWithCGImage:(CGImageRef)uiImage.CGImage
+                                scale:uiImage.scale
```
Comment (KazuCocoa) [inline WebDriverAgentLib/Utilities/FBImageIOScaler.m:141]: Should here be `1.0`?
I ran the failing view test with WDA 3.4.0, then it was passed. Main point was different image width/height.
https://dev.azure.com/AppiumCI/Appium%20CI/_build/results?buildId=14589&view=results

diff: https://github.com/appium/appium-xcuitest-driver/compare/get-viewport-screenshot-check?expand=1

(I haven't tested with this 1.0)

or around `[uiImage drawInRect:CGRectMake(0, 0, scaledSize.width, scaledSize.height)];`?

Comment (mykola-mokhnach) [reply WebDriverAgentLib/Utilities/FBImageIOScaler.m:141]: I did check the actual screenshot file resolution and it was ok. Perhaps, there is a bug in the images parsing/comparison? I assume the image library there must properly handle scaling if it is present in the image metadata.

Comment (KazuCocoa) [reply WebDriverAgentLib/Utilities/FBImageIOScaler.m:141]: mm, thanks for the checking.
noted https://github.com/appium/WebDriverAgent/issues/487

**Pull request merged**